File: Candy.py
Candy: -


The solution in Python: -


from itertools import *
  
class Solution:
    def candy(self, ratings: List[int]) -> int:
        n = len(ratings)
        dp = [1] * n
        
        for i in range(1, n):
            if ratings[i] > ratings[i - 1] and dp[i] <= dp[i - 1]:
                dp[i] = dp[i - 1] + 1

        for i in range(n - 2, -1, -1):
            if ratings[i] > ratings[i + 1] and dp[i] <= dp[i + 1]:
                dp[i] = dp[i + 1] + 1

        return sum(dp)

File: Candy.cpp
Candy: -


The solution in CPP: -


class Solution {
public:
    int candy(vector<int>& a) {
        int n = a.size();
        vector<int>v(n,1);
        for(int i=1;i<n;i++)
        {
            if(a[i]>a[i-1])
            {
                v[i] = v[i-1] + 1;
            }
        }
        for(int i=n-2;i>=0;i--)
        {
            if(a[i]>a[i+1])
            {
                v[i] = max(v[i],v[i+1]+1);
            }
        }
        int ans = 0;
        for(auto i:v)
        {
            ans += i;
        }
        return ans;      
    }
};

